### PR TITLE
DBZ-1163 Enable postgres snapshot for tables without primary key

### PR DESCRIPTION
**Pull request by preethi29**

### CODE REVIEW

**Comment by preethi29** [comment]
any comments or updates on this PR? @jpechane

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java [hunk 1]
```diff
@@ -392,8 +392,7 @@ protected void generateReadRecord(TableId tableId, Object[] rowData) {
         Object key = tableSchema.keyFromColumnData(rowData);
         Struct value = tableSchema.valueFromColumnData(rowData);
 
-        //note this is different than implementation in Streams producer. See DBZ-1163
-        if (key == null || value == null) {
+        if ( value == null) {
```
Comment (jpechane) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java]: Superfluous space after `(`

**Comment by gunnarmorling** [comment]
Thanks a lot, @preethi29!

It'd be good to have a test for this in `RecordsStreamProducerIT`, too. Another question is what to use as message key in this case:

1. simply `null`
1. one (the first) unique constraint for that table

IIRC, we do 2. for MySQL. Perhaps you could double check on this and see whether the same could be done for Postgres? @jpechane, do you know from the top of your head how it is for SQL Server? It'd be nice to have a consistent behavior for all the (relational) connectors eventually.

**Comment by gunnarmorling** [comment]
Ok, so we indeed use the first unique column as message key for MySQL. Let's wait to hear back from @jpechane about SQL Server. I think we should do the same for PG.

**Comment by jpechane** [comment]
@gunnarmorling Both SQL Server on Oracle drops non-key messages. MySQL requires either key or value to be non null to send the message.

**Comment by preethi29** [comment]
@jpechane @gunnarmorling What scenario should we test on `RecordsStreamProducerIT`

About the key, I am little lost with searching the code for this behaviour in MYSQL. Can you point me to the right code?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java (hunk 1 above)
Comment (preethi29) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java]: this is fixed

**Comment by preethi29** [comment]
The current behaviour is to send key as null in postgres. One other thing to note is, key is `null` when the table has replica identity configured and no primary key.

**Comment by gunnarmorling** [comment]
Thanks a lot, @preethi29! I've rebased and applied this fix. I've also filed [DBZ-1225](https://issues.jboss.org/browse/DBZ-1225) as a follow-up, so to make this behavior in case of null keys consistent across all the relational Debezium connectors. If you'd like to help with that, e.g. for the Postgres connector, please let us know :)

Finally, I didn't have your last name, so I've just added you as "Preethi S" to the list of all contributors in _COPYRIGHT.txt_. If you give me the full name, I can add that instead. Thanks again!

**Comment by preethi29** [comment]
Thanks @gunnarmorling.  Sure I can help with DBZ-1225 for postgres connector.
My full name is 'Preethi Sadagopan'.

**Comment by gunnarmorling** [comment]
>  Sure I can help with DBZ-1225 for postgres connector.

Thanks, that sounds great! I'd suggest you add a comment to that issue in JIRA then so that others know you're working on it. If you run into any issues, please let us know.

**Pull request closed without merging**